Standards/scs-0215-v1-robustness-features.md

```diff
@@ -0,0 +1,268 @@
+accessed by a distributed system or cluster of machines. For these reasons, etcd was chosen as the default database
+for Kubernetes.
+In order to remain reliable, an etcd cluster needs periodic maintenance. This is necessary to maintain the etcd keyspace;
+failure to do so could lead to a cluster-wide alarm, which would put the cluster into a limited-operation mode.
+To mitigate this scenario, the etcd keyspace can be compacted. Additionally, an etcd cluster can be defragmented, which
+gives back disk space to the underlying file system and can help bring the cluster back into an operable state, if it
+ran out of space earlier.
+
+etcd keyspace maintenance can be achieved by providing the necessary flags/parameters to etcd, either via the KubeadmControlPlane or in the
+configuration file of the etcd cluster, if it is managed independent of the Kubernetes cluster.
+Possible flags, that can be set for this feature, are:
+
+* auto-compaction-mode
+* auto-compaction-retention
+
+More information about compaction can be found in the respective etcd documentation
+[etcd maintenance](https://etcd.io/docs/v3.3/op-guide/maintenance/)
+
+### etcd backup
+
+An etcd cluster should be regularly backed up in order to be able to restore the cluster to a known good state at an
+earlier space in time if a failure or incorrect state happens.
+The cluster should be backed up multiple times in order to have different possible states to go back to. This is especially
+useful, if data in the newer backups was already corrupted in some way or important data was deleted in them.
+For this reason, a backup strategy needs to be developed with a decreasing number of backups in an increasing period of time,
+meaning that the previous year should only have 1 backup, but the current week should have multiple.
+Information about the backup process can be found in the etcd documentation:
+[Upgrade etcd](https://kubernetes.io/docs/tasks/administer-cluster/configure-upgrade-etcd/)
+
+### Certificate rotation
+
+In order to secure the communication of a Kubernetes cluster, (TLS) certificates signed by a controlled
+Certificate Authority (CA) can be used.
+Normally, these certificates expire after a set period of time. In order to avoid expiration and failure of a cluster,
+these certificates need to be rotated regularly and at best automatically.
+Certificates can either be rotated manually (a reference for manually working with certificates can be found
+[here](https://github.com/kelseyhightower/kubernetes-the-hard-way/blob/79a3f79b27bd28f82f071bb877a266c2e62ee506/docs/04-certificate-authority.md))
+or automatically, which requires other things to care about in a deployment.
+
+Some tools or clusters provide possibilities to rotate certificates manually.
+For example, `kubeadm` and `k3s` provides the following commands
+
+```bash
+# kubeadm
+kubeadm certs renew all
+
+# k3s
+k3s certificate rotate
```

Comment:
```suggestion
For example, `kubeadm` provides the following command

```bash
# kubeadm
kubeadm certs renew all
```
Remove k3s to avoid confusion since we don't use this in KaaS

Reply:
I can do that, but I thought more about this a general example. Maybe "we" don't use it, but what if someone external wants to make his k3s cluster SCS conform?
Or do we generally say, that this isn't a thing?

Reply:
@cah-hbaum ok let's keep it. Since there is a k3s providing Kubernetes on IaaS, it's probably not too far away from us.

Reply:
Dont know how to "resolve" your changes if we don't integrate them, maybe you need to approve or something similar.